### PR TITLE
Refacto GuestTrackingController

### PR DESCRIPTION
**Pull request by julienbourdeau**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Refactor controller, improve design, fix fatals. |
| Type? | refacto |
| Category? | FO |
| BC breaks? | no |
| Deprecations? | no |
| How to test? | Make order as guest and click link in email or go to `/index.php?controller=guest-tracking` |
### TODO
- [x] Refactor Controller
- [x] Add style
- [x] Readd order check
- [x] Make sure the guest is transformed into a customer
- [x] Edit email link -----> https://github.com/PrestaShop/emailgenerator/pull/13
- [x] Add selenium test since this page is rarely tested

### CODE REVIEW

File: controllers/front/GuestTrackingController.php [hunk 1]
```diff
@@ -23,125 +23,133 @@
...
+                'Please log in to your customer account to view the order',
+                array(),
+                'Shop.Notifications.Info'
+            );
+            $this->redirectWithNotifications($this->context->link->getPageLink('history'));
+        } else {
+            $this->order = Order::getByReferenceAndEmail($order_reference, $email);
+            if (!Validate::isLoadedObject($this->order)) {
+                $this->errors[] = $this->getTranslator()->trans(
+                    'We cound\'t find your order with the information provided, please try again',
```
Comment (Quetzacoalt91) [inline controllers/front/GuestTrackingController.php]: `couldn't` ?

File: controllers/front/GuestTrackingController.php [hunk 2]
```diff
@@ -23,125 +23,133 @@
...
-            sleep(1);
+        if (!Validate::isLoadedObject($this->order)) {
+            return $this->setTemplate('customer/guest-login.tpl');
         }
 
-        if (count($this->order_collection) > 0) {
-            $this->setTemplate('customer/guest-tracking.tpl');
-        } else {
-            $this->setTemplate('customer/guest-login.tpl');
+        if ((int) $this->order->isReturnable()) {
```
Comment (Quetzacoalt91) [inline controllers/front/GuestTrackingController.php]: What are the possible values ? Something else than a boolean ?

Comment (julienbourdeau) [inline controllers/front/GuestTrackingController.php]: Yeah it's either false or the number of day left for the return
https://github.com/julienbourdeau/PrestaShop/blob/0bf5e8706c800171060819a16d23227a7efe1a32/classes/order/Order.php#L1161-L1161

File: controllers/front/GuestTrackingController.php (hunk 1 above)
Comment (julienbourdeau) [inline controllers/front/GuestTrackingController.php]: lol

**Comment by vTerenti** [comment]
Hi!

Thanks for your contribution!

Best regards

**Pull request merged**